crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -949,6 +949,7 @@ impl CheckpointBuilder {
             let mut pending = HashSet::new();
             for effect in roots {
                 let digest = effect.transaction_digest();
+                seen.insert(*digest);
```

Comment:
This seem like a significant change (not just metrics). Can you give more context?

Reply:
Added a comment. IIUC, the builder does not need to try to get effects that are roots, since roots already get pushed into results. But lmk if I misunderstood.

Reply:
Sorry, I actually don't seem to see a comment (forgot to push?)

Reply:
I think this is ok, but I am a still a bit worried about us slipping in changes like this...

Reply:
Yeah, I thought it is a small change but will break it out to a separate PR next time ...